pyk/src/tests/integration/test_smt_hooks.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pytest
+
+from pyk.kore.prelude import BOOL, INT, TRUE, eq_int, int_dv
+from pyk.kore.rpc import SatResult
+from pyk.kore.syntax import App, Equals, EVar
+from pyk.testing import KoreClientTest
+
+if TYPE_CHECKING:
+    from typing import Final
+
+    from pyk.kore.rpc import KoreClient
+
+
+T_DIVISION_TEST_DATA: Final = (
```

Comment:
Please add a test that uses this same table to check the LLVM hook implementations.

Here's a similar (though a bit more complex) test as an example: https://github.com/runtimeverification/k/blob/42e03545c319847e7df84e6fbc196b2f51c148a0/pyk/src/tests/integration/test_krun.py#L87

Reply:
I have added a second test that uses `simplify`. This will call the HS-internal hooks for legacy kore and the LLVM library for booster.

Reply:
Nice, this way HS hooks are covered as well.

Please still add a test that calls `interpreter` directly. Then regardless of how `simplify` is implemented, we can be sure the LLVM backend hooks are exercised (even if there's a regression in Booster).

Reply:
Personally, I think this might be a bit of a scope creep...  checking the consistency of different backends was not the purpose of my change at all (it was just consistently wrong wrt. the SMT hook).
I can implement another test but that should be in the `test_krun.py` file where everything is set up for it. The tests I added inherit from `KoreClientTest` which implies a _Haskell_ backend compilation.

Reply:
> I think this might be a bit of a scope creep

I don't think so, we need to ensure that all implementations of these operations agree on the semantics.

But since the hooks are already exercised through Booster `simplify`, I can approve this PR right now if you prefer. In that case, please add an issue for this so that I can add the `interpret` tests later.

Also, before merging, please test with `evm-semantics` and `kontrol` to see if there's a regression due to the changes in `smt-hook`.

> I can implement another test but that should be in the `test_krun.py` file

Tests should be organized by what they test, and here, we're not testing `krun`, but whether `smt-hook` and `hook` agree on their semantics. Accordingly, the whole test data table and the definition test can be shared between the two tests.

> where everything is set up for it

There's not much to set up I think, you just need to add a new `KompiledTest` with the same string for the definition, and add a test method that calls `interpret`.

Reply:
Yes, probably not much to set up... but it takes me a long time to learn what it is that needs to be set up. you are probably in a better position to do that (I just tried for an hour or so but don't fully grasp the hierarchy of tests and what they can do).
So I'd prefer if you could approve this PR and we merge it, for efficiency.

Reply:
Makes sense, approved.